### PR TITLE
Add style to plots in drag and drop mode

### PR DESCRIPTION
**Pull request by sroy3**
# 3/3 main <- #4934  <- #4942 <- this

Demo

https://github.com/iterative/vscode-dvc/assets/3683420/e1a2d775-ebc9-4d19-8f98-309cbedeb0a0

Part of #4867

### CODE REVIEW

File: extension/src/experiments/columns/paths.ts [hunk 1]
```diff
@@ -1,17 +1,15 @@
 import { join } from 'path'
```
Comment (sroy3) [inline extension/src/experiments/columns/paths.ts:1]: This import is what why I need to move out all constants to use them in the webview.

File: webview/src/plots/components/util.ts [hunk 2]
```diff
@@ -43,3 +48,22 @@ export const changeDragAndDropMode = (
       : toggleCustomDragAndDrop
   return dispatch(toggleMode(!isDragAndDropMode))
 }
+
+const cleanTitlePart = (title: string) => {
```
Comment (sroy3) [inline webview/src/plots/components/util.ts:52]: There is a bit of duplication between this and `splitColumnPath`, but it's not possible to import this function inside the webview.

**Comment by mattseddon** [comment]
Found a bug with what seems like the drag end event. Clicking without moving leaves the plots in d'n'd mode:

https://github.com/iterative/vscode-dvc/assets/37993418/9814f31d-58e0-496a-ba4b-94e929a8b715

**Comment by mattseddon** [comment]
WDYT about adding a border to the plot whilst in d'n'd mode? Something like:

image

Although, I think having a border in-place of where the canvas was would look better.

Just a suggestion.

**Comment by sroy3** [comment]
> Found a bug with what seems like the drag end event. Clicking without moving leaves the plots in d'n'd mode:
> 
>  Screen.Recording.2023-11-10.at.8.52.39.am.mov

I stumbled upon it too and forgot to fix it, thanks.

**Comment by sroy3** [comment]
> WDYT about adding a border to the plot whilst in d'n'd mode? Something like:
> 
> image
> Although, I think having a border in-place of where the canvas was would look better.
> 
> Just a suggestion.
> 
> 

I'll play with it a little more.

For the border around the whole plot was making the whole thing way too crowded. It might be subtle with your theme, but they're not all like that. It also makes it harder to spot the drop target as it's not popping as much as before.

I'm also not sure if it's a good idea to give more importance to decorative elements rather than the identification of the plot itself.

File: webview/src/plots/components/DragAndDropPlot.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  plot,
+  sectionKey
+}) => {
+  const { spec, isTemplatePlot } = useGetPlot(sectionKey, plot)
+
+  let title = spec?.titles.main.normal as unknown as string
+  let subtitle = ''
+
+  if (!isTemplatePlot) {
+    const yTitle = spec?.titles.y.normal as unknown as string
```
Comment (mattseddon) [inline webview/src/plots/components/DragAndDropPlot.tsx:26]: [F] We'll be able to get these for all plots with the change in #4734 (more conflicts 😭)

**Review by mattseddon — APPROVED** [review]

File: webview/src/plots/components/styles.module.scss [hunk 4]
```diff
@@ -178,6 +197,7 @@ $gap: 20px;
 
 .smallDropIcon {
```
Comment (julieg18) [inline webview/src/plots/components/styles.module.scss:198]: Probably just me staring at things too long, but the icon seemed... out of place 🤔. One idea I played with was having the drag and drop icon be centered like targets, but with less opacity, keeping the title the focus: 

https://github.com/iterative/vscode-dvc/assets/43496356/fe2d6908-3cd3-493d-af3a-abce05bf5d27

What do you think?

**Review by julieg18 — APPROVED** [review]
Great work!

**Comment by sroy3** [comment]
New demo


https://github.com/iterative/vscode-dvc/assets/3683420/7383f349-0b27-4a9e-a4c8-17ba6d6907ee

**Pull request merged**